### PR TITLE
rewrite in es6 and add more test coverage

### PR DESCRIPTION
**Pull request by archlichking**
1. rewrite in es6
2. more unit test coverage
3. separate executor

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
-    "magellan:setup": "exit 0",
-    "magellan:teardown": "exit 0"
+    "build": "./node_modules/.bin/babel src --out-dir lib --source-maps && cp -r injections lib/",
+    "build:test:clean": "rm -rf tests/lib/* && ./node_modules/.bin/babel tests/src --out-dir tests/lib --source-maps",
+    "build:clean": "rm -rf temp/* && rm -rf reports/* && rm -rf lib/* && npm run build",
+    "test:sauce": "./node_modules/.bin/magellan --external_build_id=$TRAVIS_BUILD_ID --sauce --bail_time=${MAGELLAN_INDIVIDUAL_TEST_BAIL_TIME} --max_workers=${MAGELLAN_MAX_WORKERS} --browsers=${TIER_1_BROWSERS},${TIER_2_BROWSERS} --nightwatch_config tests/conf/nightwatch.json --sync_browsers safari:10,iphone,ipad",
+    "test": "npm run build && npm run build:test && ./node_modules/.bin/mocha --recursive tests/lib",
+    "coverage": "npm run build:clean && npm run build:test:clean && ./node_modules/.bin/istanbul cover ./node_modules/mocha/bin/_mocha -- --recursive tests/lib",
+    "upload-coverage" : "./node_modules/.bin/codecov --root=lib",
+    "prepublish": "npm run build:clean"
```
Comment (jherr) [inline package.json]: Can we add linting as well?

File: src/assertions/elContainsText.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+"use strict";
+
+import util from "util";
+import clc from "cli-color";
+
+import selectorUtil from "../util/selector";
+import BaseAssertion from "../base-assertion";
+
+let ElContainsText = function (nightwatch = null, customized_settings = null) {
```
Comment (jherr) [inline src/assertions/elContainsText.js]: This could turn into an ES6 class.

File: src/assertions/elContainsText.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  BaseAssertion.call(this, nightwatch, customized_settings);
+  this.cmd = "elcontainstext";
+}
+
+util.inherits(ElContainsText, BaseAssertion);
+
+ElContainsText.prototype.assert = function (actual, expected) {
+  let pactual = actual.replace(/[\s|\n]+/g, ' ');
```
Comment (jherr) [inline src/assertions/elContainsText.js]: I think this should be a `const`, which is something linting would pick up.

File: src/assertions/elContainsText.js (hunk 2 above)
Comment (archlichking) [inline src/assertions/elContainsText.js]: all commands and assertions cannot be es6 class. nightwatchjs wont read it as all es6 class methods are not enumerable.

File: package.json (hunk 1 above)
Comment (archlichking) [inline package.json]: yeah let me add it

**Pull request closed without merging**